optuna/storages/_rdb/alembic/versions/v2.4.0.a.py

```diff
@@ -0,0 +1,126 @@
+"""empty message
+
+Revision ID: v2.4.0.a
+Revises: v1.3.0.a
+Create Date: 2020-11-17 02:16:16.536171
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+from sqlalchemy.engine.reflection import Inspector
+from sqlalchemy.exc import SQLAlchemyError
+from sqlalchemy import orm
+
+from optuna.storages._rdb.models import StudyModel
+from optuna.storages._rdb.models import StudyDirectionModel
+from optuna.storages._rdb.models import TrialIntermediateValueModel
+from optuna.storages._rdb.models import TrialModel
+from optuna.storages._rdb.models import TrialValueModel
+from optuna.study import StudyDirection
+
+
+# revision identifiers, used by Alembic.
+revision = "v2.4.0.a"
+down_revision = "v1.3.0.a"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    bind = op.get_bind()
+    tables = Inspector.from_engine(bind).get_table_names()
+
+    if "study_directions" not in tables:
+        op.create_table(
+            "study_directions",
+            sa.Column("study_direction_id", sa.Integer(), nullable=False),
+            sa.Column(
+                "direction",
+                sa.Enum("NOT_SET", "MINIMIZE", "MAXIMIZE", name="studydirection"),
+                nullable=False,
+            ),
+            sa.Column("study_id", sa.Integer(), nullable=True),
+            sa.Column("objective", sa.Integer(), nullable=True),
```

Comment:
Should these fields really be nullable, or is it only for the migration?

Reply:
These fields are automatically generated by alembic. I confirmed that unit tests and migration tests are passed with `nullable = False`. Should we replace `nullable=True` as `nullable=False`?

Reply:
I'm wondering if this shouldn't be specified at the model layer.